zmq/test/zmq_test.ml

```diff
@@ -82,20 +82,20 @@ let test_socket_options () =
 let test_monitor () =
   let ctx = Context.create () in
   let endpoint = "tcp://127.0.0.1:51234" in
-  let s1 = ZMQ.Socket.create ctx ZMQ.Socket.pair
-  and s2 = ZMQ.Socket.create ctx ZMQ.Socket.pair in
-  let m1 = let mon_t = ZMQ.Monitor.create s1 in
-           ZMQ.Monitor.connect ctx mon_t
-  and m2 = let mon_t = ZMQ.Monitor.create s2 in
-           ZMQ.Monitor.connect ctx mon_t
+  let s1 = Zmq.Socket.create ctx Zmq.Socket.pair
+  and s2 = Zmq.Socket.create ctx Zmq.Socket.pair in
```

Comment:
While touching these lines anyway the `and` can be replaced by `let` since the bindings are not mutually recursive.

Reply:
Its actually the exact opposite. Using the 'and' here indicates that the two bindings are totally independent and no strict ordering is needed.

Reply:
Then it is inconsistent, because `endpoint` before is also independent and has no strict ordering requirement, yet is not attached to the `let` chain via `and`.

To me `and` signalizes "there is some recursiveness going on that would break without the and" and I need to take a bit more caution looking at the line, this is not the case here.